code/src/pages/MovieDetails.js

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useState } from 'react'
+import { useParams, useHistory, Link } from 'react-router-dom'
+
+import { MOVIE_URL, POSTER_URL, BACKDROP_URL } from '../utils/urls'
+import Loader from '../components/Loader'
+import HomePath from "../components/HomePath"
+
+const MovieDetails = () => {
+    const { id } = useParams()
+    const history = useHistory()
+    const [movieDetails, setMovieDetails] = useState()
+
+    useEffect(() => {
+        fetch(MOVIE_URL(id))
+            .then(res => res.json())
+            .then(data => {
+                if (data.status_message) {
+                    history.push("/notfound")
```

Comment:
nice, but I think `.replace()` would be better because it will prevent returning to a bad URL by clicking the back key of your browser.